### ISSUE TITLE
What does the `--cache-copy-layers` flag do?

### ISSUE THREAD

**Issue by tal-zvon** [GoogleContainerTools/kaniko#1940]
This isn't exactly a bug report. Just a question, with maybe a suggestion for some additional documentation.

According to the [caching section of the kaniko README](https://github.com/GoogleContainerTools/kaniko/blob/main/README.md#caching):

> kaniko can cache layers created by `RUN` and `COPY` (configured by flag `--cache-copy-layers`) commands in a remote repository

This makes it sound like either all `RUN` and `COPY` commands are always cached, or they're not, depending on if `--cache-copy-layers` is set.

Most people reading this are probably already aware that according to [docker's cache page](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache), `RUN` commands are always cached, as long as the command itself in the dockerfile hasn't changed, and `COPY` commands store file hashes along with the image, and only invalidate the cache if the files being copied changed.

Is this how kaniko works WITHOUT the `--cache-copy-layers` flag? Is this how kaniko works WITH the `--cache-copy-layers` flag? Or does Kaniko work completely differently than how docker works, and either enables or disables caching of `RUN` and `COPY` commands, without using file hashes for the `COPY` command, or string comparisons for the `RUN` command?

**Comment by korenyoni** [comment]
Agreed, the README could use some elaborating here. A simple clause explaining the behaviour _without_ this flag would clarify things quite a bit.

In terms of what this flag actually does, there is context https://github.com/GoogleContainerTools/kaniko/pull/1408 and https://github.com/GoogleContainerTools/kaniko/pull/1518.

I haven't taken a super deep look, but it looks like the flag only has to do with the `COPY` command, no? I think the README may be misleading / inaccurate in mentioning `RUN` in the context of  `--cache-copy-layers`. In fact the flag name suddenly makes a lot more sense if it's only talking about `COPY` and not `RUN`.

It's probably just a grammatical issue. If you look at [this](https://github.com/mvasin/kaniko/commit/b04399eeac3da9533a8cfa1db7650b8899af5a8d) commit, it probably meant `kaniko can cache layers created by RUN (and COPY, configured by flag --cache-copy-layers) commands in a remote repository` instead of `kaniko can cache layers created by RUN and COPY (configured by flag --cache-copy-layers) commands in a remote repository` (notice the location of the left bracket).
